webapp/src/pages/Map.tsx

```diff
@@ -354,10 +393,59 @@ function MapPage() {
     )
   }, [geoMsgCoordinates, activeLayers, addGeoMsgPoint, previewPoint])
 
-  // Effect for handling point source updates
+  // Effect for handling polygon updates moove-ai-layer
+  useEffect(() => {
+    if (!activeLayers.includes('moove-ai-layer')) return
+    const pointSourceFeatures: Array<GeoJSON.Feature<GeoJSON.Geometry>> = []
+
+    mooveAiCoordinates.forEach((point) => {
+      pointSourceFeatures.push(createPointFeature(point))
+    })
+
+    setMooveAiPolygonPointSource((prevPointSource) => ({
+      ...prevPointSource,
+      features: pointSourceFeatures,
+    }))
+
+    // Get coordinates including preview point if it exists
+    let polygonCoords = [...mooveAiCoordinates]
+    if (previewPoint && addMooveAiPoint) {
+      const previewCoords = previewPoint.geometry.coordinates
+
+      if (polygonCoords.length >= 3 && polygonCoords[0] === polygonCoords[polygonCoords.length - 1]) {
+        // For completed polygon: Remove closing point, add preview, then close
+        polygonCoords = polygonCoords.slice(0, -1)
+        polygonCoords.push(previewCoords)
+        polygonCoords.push(polygonCoords[0])
+      } else if (polygonCoords.length === 2) {
+        // For two points: Draw triangle with preview point
+        polygonCoords.push(previewCoords)
+        polygonCoords.push(polygonCoords[0])
+      } else if (polygonCoords.length === 1) {
+        // For one point: Draw line to preview point
+        polygonCoords = [[...polygonCoords[0]], [...previewCoords]] // Create a fresh array with both points
+      }
+    } else if (polygonCoords.length >= 3) {
+      // Close the polygon if we have 3+ points and no preview
+      polygonCoords.push(polygonCoords[0])
+    }
+
+    setMooveAiPolygonSource(
+      (prevPolygonSource) =>
+        ({
+          ...prevPolygonSource,
+          geometry: {
+            type: polygonCoords.length === 2 ? 'LineString' : 'Polygon', // Use LineString for 2 points
+            coordinates: polygonCoords.length === 2 ? polygonCoords : [polygonCoords],
+          },
+        } as GeoJSON.Feature<GeoJSON.Geometry>)
+    )
+  }, [mooveAiCoordinates, activeLayers, addMooveAiPoint, previewPoint])
```

Comment:
suggestion (non-blocking) - refactor this to use useMemo -> it looks like mooveAiPolygonPointSource and mooveAiPolygonSource are only updated here, so no need to re-render unnecessarily with useEffect

Reply:
Addressed